### PR TITLE
PVF timeouts follow-up

### PR DESCRIPTION
**Pull request by m-cat**
# PULL REQUEST

## Overview

This PR is a followup implementing some nits from https://github.com/paritytech/polkadot/pull/6139 that were not addressed there.

I also went ahead and made the implementor's guide a bit more readable, since I was already touching the files.

### CODE REVIEW

File: roadmap/implementers-guide/src/pvf-prechecking.md [hunk 1]
```diff
@@ -1,32 +1,14 @@
...
-- **Preparation:** This is the process of preparing the WASM blob and includes
-  both *prevalidation* and *compilation*. As prevalidation is pretty minimal
-  right now, preparation mostly consists of compilation. Note that *prechecking*
-  just consists of preparation, whereas *execution* will also prepare the PVF if
-  the artifact is not already found.
-- **Prevalidation:** Right now this just tries to deserialize the binary with
-  parity-wasm. It is a part of *preparation*.
-- **Compilation:** This is the process of compiling a PVF from wasm code to
-  machine code. It is a part of *preparation*.
+> ⚠️ This discusses a mechanism that is currently under-development. Follow the progress under [#3211][3211].
```
Comment (m-cat) [inline roadmap/implementers-guide/src/pvf-prechecking.md:3]: The hashtag was breaking this link, this made it work.

File: roadmap/implementers-guide/src/glossary.md [hunk 2]
```diff
@@ -2,44 +2,71 @@
...
+  within a given timeout.
+- **Execution:** This actually executes the PVF. The node may not have the
+  artifact from prechecking, in which case this process also includes a
+  *preparation* job. The timeout for preparation here is more lenient than when
+  prechecking.
+- **Preparation:** This is the process of preparing the WASM blob and includes
+  both *prevalidation* and *compilation*. As prevalidation is pretty minimal
+  right now, preparation mostly consists of compilation. Note that *prechecking*
+  just consists of preparation, whereas *execution* will also prepare the PVF if
+  the artifact is not already found.
```
Comment (bkchr) [inline roadmap/implementers-guide/src/glossary.md:60]: I don't get this last sentence. prechecking == preparation && execution == preparation?

Comment (m-cat) [reply roadmap/implementers-guide/src/glossary.md:60]: Yeah that is confusing, I will try to reword it a bit. How about this: 

Prechecking *only* tries to prepare the artifact, with a strict timeout. Execution will also prepare (if the artifact is not found), but with a more lenient timeout, *in addition* to then executing the compiled code.

File: roadmap/implementers-guide/src/glossary.md [hunk 3]
```diff
@@ -2,44 +2,71 @@
...
+- **Prevalidation:** Right now this just tries to deserialize the binary with
+  parity-wasm. It is a part of *preparation*.
+- **Compilation:** This is the process of compiling a PVF from wasm code to
+  a machine code *artifact*. It is a part of *preparation*.
+- **Artifact:** The compiled machine code for the wasm module.
+
+## See Also
```
Comment (pepyakin) [inline roadmap/implementers-guide/src/glossary.md:67]: I think you get a bit carried away with your PVF hat on : ) IOW, I don't think this deserves a separate section in the glossary. This is the whole project glossary, and it's not clear if we can continue to introduce sections for each area of interest (and not sure if there can always be a clear divide).

If we do remove the section, then some overly abstract terms will have to go, specifically compilation and execution. But I won't be too sad about that, I don't think they are absolutely necessary. In my subjective opinion, they are quite understandable without explicit spelling out.

They still do care some additional information, but that information should live somewhere else. E.g. overview of a PVF execution mechanism, which does not exist right now, but seems to be useful.

Comment (m-cat) [reply roadmap/implementers-guide/src/glossary.md:46]: Fair enough, I just wanted to be thorough. :) Cleaned up the glossary.

> They still do care some additional information, but that information should live somewhere else. E.g. overview of a PVF execution mechanism, which does not exist right now, but seems to be useful.

I can do that, but can you clarify where that should go? e.g. its own page, or on the PVF Prechecking page? Should it just be about execution? I added a `Summary` section to the PVF Prechecking page, maybe that's enough.

File: roadmap/implementers-guide/src/glossary.md (hunk 2 above)
Comment (m-cat) [reply roadmap/implementers-guide/src/glossary.md]: Ended up removing this sentence.

**Review by eskimor — APPROVED** [review]

**Review by pepyakin — APPROVED** [review]

**Pull request merged**